### PR TITLE
fix(docs): v3->v4 postgres migration

### LINKED ISSUE #3023: Can't complete the upgrade to v4, issues in documentation
### Question 💬

Hey folks,

I was running through the [migration guide](https://next-auth.js.org/getting-started/upgrade-v4). Everything went smoothly, up to the point where I had to make some changes in the data model.

Here are the issues that I've faced:

1. I've renamed all fields email_verified as expected. But I had an error, because the User table still requires emailVerified. The guide is wrong, but the error was straightforward and I was able to fix it
2. There's no indication that the table verification_requests is now verification_tokens. Again, error was clear enough, I could fix it on my own
3. I can't request a new token by email. I've got the following error:

```
[next-auth][error][SIGNIN_EMAIL_ERROR] 
https://next-auth.js.org/errors#signin_email_error date/time field value out of range: "1635236284553" QueryFailedError: date/time field value out of range: "1635236284553"
    at QueryFailedError.TypeORMError [as constructor] (/Users/lionelrudaz/Sites/tasters/node_modules/typeorm/error/TypeORMError.js:9:28)
    at new QueryFailedError (/Users/lionelrudaz/Sites/tasters/node_modules/typeorm/error/QueryFailedError.js:13:28)
    at PostgresQueryRunner.<anonymous> (/Users/lionelrudaz/Sites/tasters/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:277:31)
    at step (/Users/lionelrudaz/Sites/tasters/node_modules/tslib/tslib.js:143:27)
    at Object.throw (/Users/lionelrudaz/Sites/tasters/node_modules/tslib/tslib.js:124:57)
    at rejected (/Users/lionelrudaz/Sites/tasters/node_modules/tslib/tslib.js:115:69)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:93:5) {
  name: 'CreateVerificationTokenError'
```

There's no document explaining what I should do with that. Can you help me with this?

Let me know if you need more information.

### How to reproduce ☕️

Explained in the description.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

### LINKED ISSUE #3812: Documentation - Migration to V4
### Description 🐜

There is a bug in documentation or the code isn't updated. Migration from v3 to v4 says, that the User model should have updated `emailVerified` property to `email_verified` (screenshot no.1). But in the code (screenshot no 2 and 3) there is old property.

Screnshot 1 - [URL](https://github.com/nextauthjs/next-auth/blob/50fe115df6379fffe3f24408a1c8271284af660b/src/core/lib/callback-handler.ts)
Screenshot 2- [URL](https://github.com/nextauthjs/next-auth/blob/50fe115df6379fffe3f24408a1c8271284af660b/src/adapters.ts)

### Is this a bug in your own project?

No

### How to reproduce ☕️

Check docs, screenshots and two links above

### Screenshots / Logs 📽

Screenshot 2021-12-16 at 15 19 42
Screenshot 2021-12-16 at 15 19 57
Screenshot 2021-12-16 at 15 22 17

### Environment 🖥

  System:
    OS: macOS 11.4
    CPU: (8) arm64 Apple M1
    Memory: 267.06 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.1 - ~/.nvm/versions/node/v14.17.1/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 6.14.13 - ~/.nvm/versions/node/v14.17.1/bin/npm
  Browsers:
    Chrome: 96.0.4664.110
    Firefox: 95.0
    Safari: 14.1.1
  npmPackages:
    next: 12 => 12.0.7 
    next-auth: ^4.0.5 => 4.0.5 
    react: ^17.0.2 => 17.0.2 

### Contributing 🙌🏽

Yes,  I am willing to help solve this bug in a PR

### PR DESCRIPTION
**Pull request by 1000101**
## Reasoning 💡
https://github.com/nextauthjs/next-auth/pull/3874 doesn't seem to work for `postgres`. There are mistakes and the syntax is incorrect.

I've updated the schema according to what prisma is generating for v4 (`VARCHAR(255)`->`TEXT`, `INT` -> `BIGINT` -> `TEXT`, etc.)

This PR attempts to fix at least the `postgres` portion and address minor details I've found along the way (table hasn't been renamed in `MySQL` part, etc.). 

Please note I haven't tested `MySQL` part and don't know `MySQL` syntax, but it doesn't seem right either. I've also at least changed `VARCHAR(191)`->`VARCHAR(255)` as per [this](https://next-auth.js.org/adapters/prisma#create-the-prisma-schema) note:

> When using the MySQL connector for Prisma, the [Prisma String type](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#string) gets mapped to varchar(191) which may not be long enough to store fields such as id_token in the Account model. This can be avoided by explicitly using the Text type with @db.Text.

## Checklist 🧢

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## Affected issues 🎟
Hopefully closes https://github.com/nextauthjs/next-auth/issues/3023  and puts all poor souls from that issue to rest **F** 
Closes https://github.com/nextauthjs/next-auth/issues/3812

Also covers this minor migration https://github.com/nextauthjs/next-auth/pull/777

### CODE REVIEW

File: docs/docs/getting-started/upgrade-to-v4.md [hunk 1]
```diff
@@ -466,51 +467,78 @@ DROP COLUMN "updated_at"
...
-
-/* USER */
-ALTER TABLE users
-RENAME COLUMN "email_verified" "emailVerified"
-/* The following two timestamp columns have never been necessary for NextAuth.js to function, but can be kept if you want */
-DROP COLUMN "created_at"
-DROP COLUMN "updated_at"
+ALTER TABLE accounts
+ADD COLUMN IF NOT EXISTS "oauth_token_secret" TEXT NULL,
+ADD COLUMN IF NOT EXISTS "oauth_token" TEXT NULL;
```
Comment (balazsorban44) [inline docs/docs/getting-started/upgrade-to-v4.md:515]: These aren't necessary, only if you use Twitter. Should not be needed.

File: docs/docs/getting-started/upgrade-to-v4.md [hunk 2]
```diff
@@ -466,51 +467,78 @@ DROP COLUMN "updated_at"
...
 /* Note: These are only needed if you're going to be using the old Twitter OAuth 1.0 provider. */
-ADD COLUMN "oauth_token_secret" text NULL
-ADD COLUMN "oauth_token" text NULL
-
-/* USER */
-ALTER TABLE users
-RENAME COLUMN "email_verified" "emailVerified"
-/* The following two timestamp columns have never been necessary for NextAuth.js to function, but can be kept if you want */
-DROP COLUMN "created_at"
-DROP COLUMN "updated_at"
```
Comment (balazsorban44) [inline docs/docs/getting-started/upgrade-to-v4.md:496]: Why is this gone? As the comment states. These were never necessary for NextAuth.js to function

Comment (1000101) [reply docs/docs/getting-started/upgrade-to-v4.md]: It's not, I just moved it [here](https://github.com/nextauthjs/next-auth/pull/4039/files/56508d8eb58af8bd0e8d3bde8ba8ff5b2013ecba#diff-c85866e9d7e1d368d28f34725270e5f6e2e3f502f05f069c99dac996fded729fR479) due to some constraints, but that wasn't necessary, so I moved it back and committed the change.

File: docs/docs/getting-started/upgrade-to-v4.md (hunk 1 above)
Comment (1000101) [reply docs/docs/getting-started/upgrade-to-v4.md:502]: Not sure what should I do in that case. Leave it there and just comment it out?

**Comment by 1000101** [comment]
Another thing is that `expires` and `emailVerified` are now stored as `TEXT` (but apparently stored as bigint number) but previously were `TIMESTAMPTZ`... so they have to be converted during migration from `TIMESTAMPTZ`->`BIGINT`->`TEXT`.

**Comment by 1000101** [comment]
Is `expires_at` being `BIGINT` in ORM for v4 a mistake or is it intended?

**Comment by balazsorban44** [comment]
`expires_at` is a number as it's forwarded as-is from https://github.com/panva/node-openid-client/blob/main/docs/README.md#new-tokensetinput

**Comment by 1000101** [comment]
> `expires_at` is a number as it's forwarded as-is from https://github.com/panva/node-openid-client/blob/main/docs/README.md#new-tokensetinput

In that case I think this PR is ready for testing. Too bad it only works for PostgreSQL. I'm not sure if it's good that MySQL migration steps are provided but not functional.

**Comment by 1000101** [comment]
Updated & rebased on main

**Comment by 1000101** [comment]
Successfully verified migration of a v3 installation using copy&paste of postgres section.

File: docs/docs/getting-started/upgrade-to-v4.md (hunk 1 above)
Comment (1000101) [reply docs/docs/getting-started/upgrade-to-v4.md]: I've commented it out https://github.com/nextauthjs/next-auth/pull/4039/files#diff-c85866e9d7e1d368d28f34725270e5f6e2e3f502f05f069c99dac996fded729fR502-R503

**Comment by ndom91** [comment]
Hey thanks a lot for double checking the migration script and testing it out! This looks great

**Review by ndom91 — APPROVED** [review]

**Pull request merged**